src/bootstrap/config.rs

```diff
@@ -615,6 +616,8 @@ impl Config {
                 target.musl_root = cfg.musl_root.clone().map(PathBuf::from);
                 target.wasi_root = cfg.wasi_root.clone().map(PathBuf::from);
                 target.qemu_rootfs = cfg.qemu_rootfs.clone().map(PathBuf::from);
+                target.no_std =
+                    cfg.no_std.unwrap_or(triple.contains("-none-") || triple.contains("nvptx"));
```

Comment:
Hm, is `-none-` perhaps a bit too broad? I guess we presumably don't ship any platforms with that triple today that do have std compiled?

Reply:
The logic is taken from `if target.contains("-none-") || target.contains("nvptx") {`. Some later refactor should just ensure there are `Target`s for all platforms specified by name, rather than config, so we can deduplicate this.